### ISSUE TITLE
IllegalArgumentException (EMF-CCFinder)

### ISSUE THREAD

**Issue by dmazinanian** [tsantalis/JDeodorant#35]
Clone # 190 (Group 1807, Pair 1-2): org.eclipse.emf.mapping.command.AddOverrideCommand#getAffectedObjects (row 575) and org.eclipse.emf.mapping.command.AddOverrideCommand#getResult (row 576)

```
java.lang.IllegalArgumentException
    at org.eclipse.jdt.core.dom.rewrite.ListRewrite.insertLast(ListRewrite.java:249)
    at gr.uom.java.jdeodorant.refactoring.manipulators.RefactoringUtility.createParameterizedType(RefactoringUtility.java:144)
    at gr.uom.java.jdeodorant.refactoring.manipulators.RefactoringUtility.generateTypeFromTypeBinding(RefactoringUtility.java:88)
    at gr.uom.java.jdeodorant.refactoring.manipulators.ExtractCloneRefactoring.extractClone(ExtractCloneRefactoring.java:1293)
    at gr.uom.java.jdeodorant.refactoring.manipulators.ExtractCloneRefactoring.apply(ExtractCloneRefactoring.java:331)
    at gr.uom.java.jdeodorant.refactoring.manipulators.ExtractCloneRefactoring.checkFinalConditions(ExtractCloneRefactoring.java:4833)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.testRefactoring(Application.java:661)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.start(Application.java:184)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1492)
```

**Comment by dmazinanian** [comment]
I guess the same in EMF-CloneDR - Clone # 269 (Group 269, Pair 1-2): org.eclipse.emf.common.util.BasicEMap#entryForKey (row 1952) and org.eclipse.emf.common.util.BasicEMap#entryForKey (row 1953)

**Comment by tsantalis** [comment]
First report was already fixed in another issue.
Second report fixed in commit [e2f44e165cc1fa51eb54b1c2e85e300e677ca803](https://github.com/tsantalis/JDeodorant/commit/e2f44e165cc1fa51eb54b1c2e85e300e677ca803)
